cpp/src/search/contains_table.cu

```diff
@@ -408,31 +306,7 @@ rmm::device_uvector<bool> contains(table_view const& haystack,
                                    rmm::cuda_stream_view stream,
                                    rmm::mr::device_memory_resource* mr)
 {
-  // Checking for only one table is enough, because both tables will be checked to have the same
-  // shape later during row comparisons.
-  auto const has_lists = std::any_of(haystack.begin(), haystack.end(), [](auto const& col) {
-    return cudf::structs::detail::is_or_has_nested_lists(col);
-  });
-
-  if (has_lists || compare_nans == nan_equality::UNEQUAL) {
-    // We must call a separate code path that uses the new experimental row hasher and row
-    // comparator if:
-    //  - The input has lists column, or
-    //  - Floating-point NaNs are compared as unequal.
-    // Inputs with these conditions are supported only by this code path.
-    return contains_with_lists_or_nans(haystack, needles, compare_nulls, compare_nans, stream, mr);
-  }
-
-  // If the input tables don't have lists column and NaNs are compared equal, we rely on the classic
-  // code path that flattens the input tables for row comparisons. This way is known to have
-  // better performance.
-  return contains_without_lists_or_nans(haystack, needles, compare_nulls, stream, mr);
-
-  // Note: We have to keep separate code paths because unifying them will cause performance
-  // regression for the input having no nested lists.
-  //
-  // TODO: We should unify these code paths in the future when performance regression is no longer
-  // happening.
+  return contains_impl(haystack, needles, compare_nulls, compare_nans, stream, mr);

```

Comment:
If that function is called just once, please move its content here and remove it completely.

Reply:
Agreed, I don't see a need for a separate impl function here.